pdns/misc.hh

```diff
@@ -803,7 +803,7 @@ struct FDWrapper
 
   FDWrapper& operator=(FDWrapper&& rhs) noexcept
   {
-    if (d_fd != -1) {
+    if (d_fd >= 0) {
```

Comment:
should there be check here that fd != -1?

Reply:
I'm not sure I get what you have in mind?

Reply:
i was just wondering if it is acceptable, or not, to pass around closed fd handle?

Reply:
Ah, right! I believe it is fine, yes, so I don't think we need an additional check.